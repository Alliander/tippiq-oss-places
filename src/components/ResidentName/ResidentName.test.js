import React from 'react';
import renderer from 'react-test-renderer';
import Element from './ResidentName';

describe('ResidentName', () => {
  test('with displayName', () => {
    const component = renderer.create(
      <Element text={'Chuck Norris'} />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('with displayName', () => {
    const component = renderer.create(
      <Element text={'test@example.com'} />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('without displayName', () => {
    const component = renderer.create(
      <Element text={null} />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('with very long name', () => {
    const component = renderer.create(
      <Element text={'Edward van Llanfair­pwllgwyngyll­gogery­chwyrn­drobwll­llan­tysilio­gogo­goch'} />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('with short length limit', () => {
    const component = renderer.create(
      <Element truncateAt={5} text={'Edward van Llanfair­pwllgwyngyll­gogery­chwyrn­drobwll­llan­tysilio­gogo­goch'} />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('with long length limit', () => {
    const component = renderer.create(
      <Element truncateAt={40} text={'Edward van Llanfair­pwllgwyngyll­gogery­chwyrn­drobwll­llan­tysilio­gogo­goch'} />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('with ludicrous length limit', () => {
    const component = renderer.create(
      <Element truncateAt={9999} text={'Edward van Llanfair­pwllgwyngyll­gogery­chwyrn­drobwll­llan­tysilio­gogo­goch'} />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('with children', () => {
    const component = renderer.create(
      <Element text={'Chuck Norris'}>
        <a href="/pas-aan">Pas aan</a>
      </Element>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
