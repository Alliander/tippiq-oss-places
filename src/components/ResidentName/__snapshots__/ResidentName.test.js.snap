// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResidentName with children 1`] = `
<span
  className=""
>
  Chuck Norris
  <a
    href="/pas-aan"
  >
    Pas aan
  </a>
</span>
`;

exports[`ResidentName with displayName 1`] = `
<span
  className=""
>
  Chuck Norris
</span>
`;

exports[`ResidentName with displayName 2`] = `
<span
  className=""
>
  test@example.com
</span>
`;

exports[`ResidentName with long length limit 1`] = `
<span
  className=""
>
  Edward van Llanfair­pwllgwyngyll­goge...
</span>
`;

exports[`ResidentName with ludicrous length limit 1`] = `
<span
  className=""
>
  Edward van Llanfair­pwllgwyngyll­gogery­chwyrn­drobwll­llan­tysilio­gogo­goch
</span>
`;

exports[`ResidentName with short length limit 1`] = `
<span
  className=""
>
  Ed...
</span>
`;

exports[`ResidentName with very long name 1`] = `
<span
  className=""
>
  Edward van Llanfa...
</span>
`;

exports[`ResidentName without displayName 1`] = `
<span
  className=""
>
  Bewoner
</span>
`;
